File: Bailaconmigo/src/main/java/org/example/bailaconmigo/DTOs/EventRegistrationResponseDto.java
package org.example.bailaconmigo.DTOs;

import lombok.Data;
import org.example.bailaconmigo.Entities.Enum.RegistrationStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class EventRegistrationResponseDto {
    private Long id;
    private Long eventId;
    private String eventName;
    private LocalDateTime eventDateTime;
    private Long dancerId;
    private String dancerName;
    private LocalDateTime registrationDate;
    private RegistrationStatus status;
    // ===== CÓDIGO DINÁMICO =====
    private String codigoDinamico;  // NUEVO CAMPO

    // ===== CAMPOS DE PAGO =====
    private BigDecimal paidAmount;
    private BigDecimal appFee;
    private BigDecimal organizerAmount;
    private String paymentPreferenceId;
    private String paymentId;
    private String paymentStatus;
    private String paymentReference;
    private LocalDateTime paymentDate;
    private String paymentMethod;

    // Campo de asistencia
    private Boolean attended;

    // URL de pago para redirigir al usuario
    private String paymentUrl;
}

File: frontend/src/app/models/EventRegistrationResponseDto.ts
import { RegistrationStatus } from "./RegistrationStatus";

export interface EventRegistrationResponseDto {
  id: number;
  eventId: number;
  eventName: string;
  eventDateTime: Date;
  dancerId: number;
  dancerName: string;
  registrationDate: Date;
  status: RegistrationStatus;
  codigoDinamico: string;
  paidAmount: number;
  appFee: number;
  organizerAmount: number;
  paymentPreferenceId: string;
  paymentId: string;
  paymentStatus: string;
  paymentReference: string;
  paymentDate: string;
  paymentMethod: string;
  attended: boolean;
  paymentUrl: string;
}